plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy

```diff
@@ -99,44 +106,43 @@ class NomadService implements Closeable{
         JobRegisterRequest jobRegisterRequest = new JobRegisterRequest()
         jobRegisterRequest.setJob(job)
 
-        if( saveJsonPath ) try {
+        if (saveJsonPath) try {
             saveJsonPath.text = job.toString()
         }
-        catch( Exception e ) {
-            log.debug "WARN: unable to save request json -- cause: ${e.message ?: e}"
+        catch (Exception e) {
+            log.debug "[NOMAD] Unable to save request json -- cause: ${e.message ?: e}"
         }
 
 
         try {
             JobRegisterResponse jobRegisterResponse = jobsApi.registerJob(jobRegisterRequest, config.jobOpts().region, config.jobOpts().namespace, null, null)
             jobRegisterResponse.evalID
-        } catch (Throwable e) {
-            log.debug("[NOMAD] Failed to submit ${job.name} -- Cause: ${e.message ?: e}", e)
-            throw new ProcessSubmitException("[NOMAD] Failed to submit ${job.name} -- Cause: ${e.message ?: e}", e)
```

Comment:
@jagedn , this is the part where I think we should reconsider how to manage server failure. 

I guess, we should distinguish between `server` connection (`HTTP layer` ) failure and then `Nextflow layer` failure.


So, I'm double-minded whether we should keep the `ProcessSubmitException` and just add some checks on the `HTTP(s) connection` layer. 

What do you think?

Reply:
I was investigating (a little) and my idea is to add an interceptor in the API constructor to retry http layer failures 

but my first approach to do it is a little tricky. I think we need to build an OkhttpClient with interceptors and provide it in the constructor

Reply:
I see the point yes, but here's a thought about existing executor connections in Nextflow - how about using `failsafe` library for managing this specific connection/


For reference https://github.com/nextflow-io/nextflow/blob/master/plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy#L909-L924

Reply:
ok, looks nice

do you will implement it or do you want I work on it ?

Reply:
Mmm, not sure if this is urgent as a use-case to be covered. It's not really recommended to restart the server with an ongoing workflow and even if this happens, the cache should be able to rescue any previous processes  💯 

I'd rather prioritise the following 

1. Improvement of Nextflow->Nomad layer, test with more Nextflow directives 
2. Improvement of the policy setup -> once @jhaezebr and @tomiles have completed the TLS setup we can accommodate that.
3. An interesting exploration would be the  `Nextflow secrets` -> `Nomad variables` integration so that the secrets are not revealed in Nomad portal 🤓 


Let me know if this seems okay to you @jagedn , otherwise happy to pivot as per your decision :)

Reply:
will focus them in secrets<->variables integration (it sounds a very interesting feature)

Reply:
Thanks dear @jagedn  :)